### PR TITLE
MEN-3832 Mention explicitly how to use multiple ssh arguments in snapshots

### PR DESCRIPTION
**Pull request by merlin-northern**
MEN-3832 Mention explicitly how to use multiple ssh arguments in snapshots

ChangeLog:none
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: 01.Get-started/03.Deploy-a-system-update/docs.md [hunk 1]
```diff
@@ -85,13 +85,13 @@ DEVICE_TYPE="raspberrypi4"
 !!! Make sure to replace `raspberrypi4` with the specific value that you are
 !!! seeing in your setup
 
-Set `SSH_ARGS` shell variable to specify the SSH access port:
+Set `SSH_ARG` shell variable to specify the SSH access port:
```
Comment (estenberg) [inline 01.Get-started/03.Deploy-a-system-update/docs.md:88]: This is a good change to indicate that only one argument can be specified!

File: 01.Get-started/03.Deploy-a-system-update/docs.md [hunk 2]
```diff
@@ -112,13 +112,31 @@ mender-artifact write rootfs-image \
...
     -o system-v1.mender \
-    -S "${SSH_ARGS}"
+    -S "${SSH_ARG}"
 ```
 
 ! Your device is not usable while the snapshot operation is in progress. Mender
 ! will freeze the storage device during this operation in order to create a
 ! consistent snapshot.
 
+Please note that if you want to pass multiple ssh arguments, you have to specify
```
Comment (estenberg) [inline 01.Get-started/03.Deploy-a-system-update/docs.md:122]: This is not the right place I think, Get started needs to be as simple as possible - explaining how to pass multiple SSH arguments is a more advanced topic (not something you would do first time you use Mender).

Would recommend to move this to https://docs.mender.io/hosted/artifacts/snapshots, perhaps under "passing multiple arguments to SSH" towards the end.

File: 01.Get-started/03.Deploy-a-system-update/docs.md [hunk 3]
```diff
@@ -112,13 +112,31 @@ mender-artifact write rootfs-image \
...
+```bash
+mender-artifact write rootfs-image \
+    -f ssh://"${USER}@${IP_ADDRESS}" \
+    -t "${DEVICE_TYPE}" \
+    -n system-v1 \
+    -o system-v1.mender \
+    --ssh-args="-p 8122" \
+    --ssh-args="-o UserKnownHostsFile=/dev/null" \
+    --ssh-args="-o StrictHostKeyChecking=no" \
+    --ssh-args="-o PubKeyAuthentication=no"
```
Comment (kacf) [inline 01.Get-started/03.Deploy-a-system-update/docs.md:136]: I'm actually quite surprised that this works, as each one of these is being passed into SSH as one gigantic argument. I can only assume that SSH supports concatenation of short flags with their arguments, and that it removes whitespace when doing so. But this is kind of "accidentally working". The 100% correct way to pass in this argument would be with:

```
    --ssh-args="-p" \
    --ssh-args="8122" \
    --ssh-args="-o" \
    --ssh-args="UserKnownHostsFile=/dev/null" \
    --ssh-args="-o" \
    --ssh-args="StrictHostKeyChecking=no" \
    --ssh-args="-o" \
    --ssh-args="PubKeyAuthentication=no"
```

This is very ugly though. And I suppose it's not that likely that SSH will change its parsing now. So I'm fine with keep what you proposed, but just thought I'd mention it.

Comment (merlin-northern) [reply 01.Get-started/03.Deploy-a-system-update/docs.md:136]: thanks @kacf

File: 01.Get-started/03.Deploy-a-system-update/docs.md (hunk 2 above)
Comment (merlin-northern) [reply 01.Get-started/03.Deploy-a-system-update/docs.md:122]: I knew you are going to say that. I use all the power I had to write something more than just removing the 'S' from ARGS ;-)
ok will do @estenberg

Comment (merlin-northern) [reply 01.Get-started/03.Deploy-a-system-update/docs.md]: @estenberg [moved](https://github.com/mendersoftware/mender-docs/pull/1097/commits/4e37c74113d80ec257db75bb704381308efa0a78#diff-4d66535e18678ad0ab6a578d1ac23782R88).

File: 01.Get-started/03.Deploy-a-system-update/docs.md (hunk 3 above)
Comment (merlin-northern) [reply 01.Get-started/03.Deploy-a-system-update/docs.md]: @kacf I checked again, that it works, so [I left it as is](https://github.com/mendersoftware/mender-docs/pull/1097/commits/4e37c74113d80ec257db75bb704381308efa0a78#diff-4d66535e18678ad0ab6a578d1ac23782R88) but I can change it -- no problem, just let me know if you consider it crucial.

Comment (kacf) [reply 01.Get-started/03.Deploy-a-system-update/docs.md]: Let's leave it for now, at least until we get any reports about it not working. I think the readability of your version is much better, even if it only works by accident.

File: 04.Artifacts/22.Snapshots/docs.md [hunk 4]
```diff
@@ -78,9 +78,30 @@ on the golden device. For example, the command:
...
 
-mender-artifact write rootfs-image -f ssh://$USER@$ADDR \
+mender-artifact write rootfs-image -f ssh://${USER}@${ADDR} \
                                    -n artifact-name \
                                    -o snapshot-release.1.0.mender \
                                    -t device-type
 ```
 Yields the exact same artifact as above.
+
+Please note you can pass extra arguments to ssh command. To this end you have to specify
```
Comment (estenberg) [inline 04.Artifacts/22.Snapshots/docs.md:88]: Suggestions:
* "arguments to ssh command" -> "arguments to ssh"
* "To this end you have to specify" -> "To do this, specify"

File: 04.Artifacts/22.Snapshots/docs.md [hunk 5]
```diff
@@ -78,9 +78,30 @@ on the golden device. For example, the command:
...
+each one of them in a separate `-S "${SSH_ARG}"` option; for instance if you want to use ssh with
```
Comment (estenberg) [inline 04.Artifacts/22.Snapshots/docs.md:89]: The last sentence (after ;) is not necessary to write out, they can see it below. Can simplify to:

..option, e.g.:

File: 04.Artifacts/22.Snapshots/docs.md (hunk 4 above)
Comment (merlin-northern) [reply 04.Artifacts/22.Snapshots/docs.md]: sure.

File: 04.Artifacts/22.Snapshots/docs.md (hunk 5 above)
Comment (merlin-northern) [reply 04.Artifacts/22.Snapshots/docs.md]: sure.

**Pull request merged**